src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java

```diff
@@ -183,18 +194,81 @@ protected String getTicket(Parameters parameters)
             throw new InvalidArgumentException("Authorization header is required.");
         }
 
+        boolean isBasicHeader = false;
+        boolean isBearerHeader = false;
+
         final String[] authorizationParts = authorization.split(" ");
-        if (!authorizationParts[0].equalsIgnoreCase("basic"))
+        if (authorizationParts[0].equalsIgnoreCase("basic"))
+        {
+            isBasicHeader = true;
+        }
+        else if (authorizationParts[0].equalsIgnoreCase("bearer"))
+        {
+            isBearerHeader = true;
+        }
+
+        if (!isBasicHeader && !isBearerHeader)
         {
             throw new InvalidArgumentException("Authorization '" + authorizationParts[0] + "' not supported.");
         }
 
-        final String decodedAuthorisation = new String(Base64.decode(authorizationParts[1]));
-        Authorization authObj = new Authorization(decodedAuthorisation);
-        if (!authObj.isTicket())
+        if (isBasicHeader)
+        {
+            final String decodedAuthorisation = new String(Base64.decode(authorizationParts[1]));
+            Authorization authObj = new Authorization(decodedAuthorisation);
+            if (!authObj.isTicket())
+            {
+                throw new InvalidArgumentException("Ticket base authentication required.");
+            }
+            return authObj.getTicket();
+        }
+        else
```

Comment:
Will it be better to use the following structure?
```
else if(isBearerHeader)
{
...
}
else
{
    throw new InvalidArgumentException("Authorization not supported.");
}
 ```

Reply:
I fixed that. Thanks